server/tests/tasks.test.ts

```diff
@@ -65,19 +64,16 @@ describe('Test /roadmaps/:roadmapId/tasks/ api', function () {
       expect(added.createdByUser).to.equal(userId);
     });
     it('Should not add new task with incorrect permissions', async function () {
+      await removePermission(Permission.RoadmapReadUsers);
+      await removePermission(Permission.TaskCreate);
```

Comment:
These should be at the same position as the previous permission patching.

Now the `before` result is also fetched with reduced permissions.

This should probably be done in each of these cases.


Maybe this could be done easier with *kind of* context manager

Instead of
```ts
await removePermission(permission);
const res = /* whatever query */
/* remember to restore previous permission */
```

```ts
const res = await withoutPermission(permission, () => {
  return /* whatever query */
});
```

The implementation would be relatively simple:

```ts
export const withoutPermission = async <T>(
  permission: Permission,
  work: () => Promise<T>,
) => {
  const firstRoadmapId = (await Roadmap.query().first()).id;
  const userId = (
    await User.query().where({ username: 'AdminPerson1' }).first()
  ).id;

  const role = await Role.query().findById([userId, firstRoadmapId]);
  const original = role.type;
  await role.$query().patch({ type: role.type & ~permission });
  try {
    return await work();
  } finally {
    await Role.query().patchAndFetchById([userId, firstRoadmapId], {
      type: original,
    });
  }
};
```

This would also work for removing multiple permissions by:
- combining permissions to remove `a | b | c`
- or nesting `withoutPermission(a, () => withoutPermission(b, () => /* ... */))` (arguably not so nice)